### ISSUE TITLE
Error after upgrading to Go 1.8

### ISSUE THREAD

**Issue by syifan** [onsi/ginkgo#324]
I recently upgraded my Go package to the 1.8 version. Ginkgo works all fine with Go 1.7. But after upgrading, I get the following error information if I run `ginkgo -r -cover` 

```
go install errors: open /usr/local/go/pkg/linux_amd64/errors.a: permission denied
go install internal/race: open /usr/local/go/pkg/linux_amd64/internal/race.a: permission denied
go install encoding: open /usr/local/go/pkg/linux_amd64/encoding.a: permission denied
go install unicode/utf16: open /usr/local/go/pkg/linux_amd64/unicode/utf16.a: permission denied
go install unicode/utf8: open /usr/local/go/pkg/linux_amd64/unicode/utf8.a: permission denied
go install internal/nettrace: open /usr/local/go/pkg/linux_amd64/internal/nettrace.a: permission denied
go install crypto/subtle: open /usr/local/go/pkg/linux_amd64/crypto/subtle.a: permission denied
go install container/list: open /usr/local/go/pkg/linux_amd64/container/list.a: permission denied
go install crypto/internal/cipherhw: open /usr/local/go/pkg/linux_amd64/crypto/internal/cipherhw.a: permission denied
go install vendor/golang_org/x/crypto/curve25519: open /usr/local/go/pkg/linux_amd64/vendor/golang_org/x/crypto/curve25519.a: permission denied
go install unicode: open /usr/local/go/pkg/linux_amd64/unicode.a: permission denied
go install math: open /usr/local/go/pkg/linux_amd64/math.a: permission denied
```

The error only happens if I enable -cover. Also `go test -cover ./...` works fine without any problem. I wonder if there is anything wrong with my setup?

**Comment by lhdomenech** [comment]
I have the same issue after upgrading to Go 1.8

**Comment by Bo0mer** [comment]
The same error occurs if the tests are run with the `--race` flag. Example:
```
$ ginkgo --race
Failed to compile dongleinfo:

go install errors: open /usr/local/go/pkg/darwin_amd64_race/errors.a: permission denied
go install internal/race: open /usr/local/go/pkg/darwin_amd64_race/internal/race.a: permission denied
go install encoding: open /usr/local/go/pkg/darwin_amd64_race/encoding.a: permission denied
go install unicode/utf16: open /usr/local/go/pkg/darwin_amd64_race/unicode/utf16.a: permission denied
go install unicode/utf8: open /usr/local/go/pkg/darwin_amd64_race/unicode/utf8.a: permission denied
go install crypto/subtle: open /usr/local/go/pkg/darwin_amd64_race/crypto/subtle.a: permission denied
go install crypto/internal/cipherhw: open /usr/local/go/pkg/darwin_amd64_race/crypto/internal/cipherhw.a: permission denied
go install internal/nettrace: open /usr/local/go/pkg/darwin_amd64_race/internal/nettrace.a: permission denied
go install container/list: open /usr/local/go/pkg/darwin_amd64_race/container/list.a: permission denied
go install vendor/golang_org/x/crypto/curve25519: open /usr/local/go/pkg/darwin_amd64_race/vendor/golang_org/x/crypto/curve25519.a: permission denied
go install unicode: open /usr/local/go/pkg/darwin_amd64_race/unicode.a: permission denied
go install math: open /usr/local/go/pkg/darwin_amd64_race/math.a: permission denied

Ginkgo ran 1 suite in 3.311995992s
Test Suite Failed
```


`go test -race` works as expected. Any chance it is related to https://github.com/golang/go/issues/19151?

Regards,
Ivan

**Comment by geraldhuff** [comment]
I had the same issue after upgrade to 1.8. I found that running `ginkgo -cover  -covermode=count` got past this issue. Can't seem to find good documentation on what mode count is vs atomic or set however.

**Comment by Bo0mer** [comment]
Basically:
* set: did each statement run?
* count: how many times did each statement run?
* atomic: like count, but counts precisely in parallel programs

More details - https://blog.golang.org/cover

**Comment by onsi** [comment]
With go 1.8.1 out today would folks mind giving that a try to see if it's helped?

**Comment by lhdomenech** [comment]
I cannot reproduce the issue with go 1.8.1.

**Comment by syifan** [comment]
The problem is not there after upgrading to go 1.8.1. Thank you very much!

**Comment by vovkasm** [comment]
I still experienced this problem.
* System: ArchLinux
* `go version go1.8.1 linux/amd64`

```sh
$ cd $PROJ
$ export GOPATH=`pwd`

$ go test woof/manager
ok      woof/manager    0.021s

$ go test -race woof/manager
ok      woof/manager    1.093s

$ ./bin/ginkgo src/woof/manager
Running Suite: manager
======================
Random Seed: 1491761640
Will run 19 of 19 specs

•••••••••••••••••••
Ran 19 of 19 Specs in 0.007 seconds
SUCCESS! -- 19 Passed | 0 Failed | 0 Pending | 0 Skipped PASS

Ginkgo ran 1 suite in 1.881389372s
Test Suite Passed

./bin/ginkgo -race src/woof/manager
Failed to compile manager:

go install errors: open /usr/lib/go/pkg/linux_amd64_race/errors.a: permission denied
go install encoding: open /usr/lib/go/pkg/linux_amd64_race/encoding.a: permission denied
go install internal/race: open /usr/lib/go/pkg/linux_amd64_race/internal/race.a: permission denied
go install internal/nettrace: open /usr/lib/go/pkg/linux_amd64_race/internal/nettrace.a: permission denied
go install crypto/internal/cipherhw: open /usr/lib/go/pkg/linux_amd64_race/crypto/internal/cipherhw.a: permission denied
go install unicode/utf16: open /usr/lib/go/pkg/linux_amd64_race/unicode/utf16.a: permission denied
go install crypto/subtle: open /usr/lib/go/pkg/linux_amd64_race/crypto/subtle.a: permission denied
go install unicode/utf8: open /usr/lib/go/pkg/linux_amd64_race/unicode/utf8.a: permission denied
go install container/list: open /usr/lib/go/pkg/linux_amd64_race/container/list.a: permission denied
go install vendor/golang_org/x/crypto/curve25519: open /usr/lib/go/pkg/linux_amd64_race/vendor/golang_org/x/crypto/curve25519.a: permission denied
go install unicode: open /usr/lib/go/pkg/linux_amd64_race/unicode.a: permission denied
go install math: open /usr/lib/go/pkg/linux_amd64_race/math.a: permission denied

Ginkgo ran 1 suite in 4.316361389s
Test Suite Failed
```

ginkgo from 13c94a130ef24c498bdb2d8d3873cf7790925dcf, so fresh enough.

Can I provide any additional info?

**Comment by onsi** [comment]
Did you `rm -RC $GOPATH/pkg` after upgrading?  Does `go test -race -I` work?

**Comment by vovkasm** [comment]
1. Yes, I deleted `$GOPATH/pkg` dir
2. Not sure, what exactly you mean (`go test -help` silent about `-I` option)...
```sh
$ go test -race -I woof/manager
can't load package: package .: no buildable Go source files in /home/vovkasm/work/server

$ cd src/woof/manager
$ go test -I
flag provided but not defined: -I
```

**Comment by onsi** [comment]
sorry, ipad autocorrect fail.  i meant

`go test -race -i` this is what ginkgo uses under the hood

**Comment by vovkasm** [comment]
Ahha, yes!
```sh
$ go test -race -i
go install errors: open /usr/lib/go/pkg/linux_amd64_race/errors.a: permission denied
...
```

**Comment by onsi** [comment]
ok weird.  I'm on OS X so perhaps there's a difference there.

I _thought_ this was supposed to be fixed in https://github.com/golang/go/issues/19151

apparently not?

**Comment by vovkasm** [comment]
Will try investigate this, in theory other factors can cause this such as timestamps on files or error in OS provided package...

On OS X most users will use go from Homebrew, then this issue will not expose self because GOROOT writeable by user.

**Comment by etsangsplk** [comment]
doe the -a options somehow trigger the go install -a?

**Comment by onsi** [comment]
which -a option?  yes `ginkgo -a` will pass the `-a` flag down to `go` and will result in slow compiles.  But I don't believe that's the issue on this thread.

I'd like to close this out - anyone seeing any reason not to?  My understanding is that the root issues are resolved in go 1.8.1+

**Comment by Bo0mer** [comment]
I do **not** have this issue any more.

Thanks.
